components/BookDetailsCard/BookDetailsCard.js

```diff
@@ -0,0 +1,96 @@
+import MyLibraryButton from "@/components/Button/MyLibraryButton";
+import AlreadyReadButton from "@/components/Button/AlreadyReadButton";
+import ReadMore from "@/components/ReadMore/ReadMore";
+import CurrentlyReadingButton from "@/components/Button/CurrentlyReadingButton";
+import styled from "styled-components";
+import Image from "next/image";
+
+export default function BookDetailsCard({
+  cover,
+  title,
+  booksInfo,
+  id,
+  author,
+  genre,
+  publishYear,
+  pages,
+  description,
+  handleToggleAlreadyRead,
+  handleToggleBookmark,
+  handleToggleCurrentlyReading,
+  pathName,
+}) {
+  return (
+    <StyledBookDetail>
+      <StyledImage
+        src={cover}
+        height={224}
+        width={150}
+        alt={`Cover Image of ${title}`}
+      />
+      {pathName === "/book-details/[id]" && (
+        <StyledButtonWrapper>
+          <MyLibraryButton
+            onToggle={handleToggleBookmark}
+            id={id}
+            isBookmarked={
+              booksInfo.find((bookInfo) => bookInfo.id === id)?.isBookmarked
+            }
+          />
+          <CurrentlyReadingButton
+            onToggle={handleToggleCurrentlyReading}
+            id={id}
+            isCurrentlyReading={
+              booksInfo.find((bookInfo) => bookInfo.id === id)
+                ?.isCurrentlyReading
+            }
+          />
+          <AlreadyReadButton
+            onToggle={handleToggleAlreadyRead}
+            id={id}
+            isAlreadyRead={
+              booksInfo.find((bookInfo) => bookInfo.id === id)?.isAlreadyRead
+            }
+          />
```

Comment:
✨ **Just Optimization**
Avoid duplicating the find operation for each button instead extract the common _bookInfo lookup_ outside of the JSX return 
```js
 const currentBookInfo = booksInfo.find(bookInfo => bookInfo.id === id);
 
 
 return (
   // ...
      {pathName === "/book-details/[id]" && (
        <StyledButtonWrapper>
          <MyLibraryButton
            onToggle={handleToggleBookmark}
            id={id}
             isBookmarked={currentBookInfo?.isBookmarked} // Use the extracted variable
          />
          <CurrentlyReadingButton
            onToggle={handleToggleCurrentlyReading}
            id={id}
            isCurrentlyReading={currentBookInfo?.isCurrentlyReading} // Use the extracted variable
          />
          <AlreadyReadButton
            onToggle={handleToggleAlreadyRead}
            id={id}
            isAlreadyRead={currentBookInfo?.isAlreadyRead} // Use the extracted variable
          />
 // ...
 ```

Reply:
✓